flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SingleLogicalSlot.java

```diff
@@ -197,20 +196,21 @@ private void signalPayloadRelease(Throwable cause) {
     }
 
     private void returnSlotToOwner(CompletableFuture<?> terminalStateFuture) {
-        terminalStateFuture.whenComplete(
-                (Object ignored, Throwable throwable) -> {
-                    if (state == State.RELEASING) {
-                        slotOwner.returnLogicalSlot(this);
-                    }
-
-                    markReleased();
-
-                    if (throwable != null) {
-                        releaseFuture.completeExceptionally(throwable);
-                    } else {
-                        releaseFuture.complete(null);
-                    }
-                });
+        FutureUtils.assertNoException(
+                terminalStateFuture.whenComplete(
```

Comment:
hmm...AFAIK this will never be completed exceptionally, but then we also shouldn't handle Throwables like we do here (and instead fail with an IllegalStateException I suppose).

Reply:
Yes, we should fail hard in this case.